src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -18,7 +19,7 @@ namespace Microsoft.CST.OpenSource.PackageManagers
     using System.Text.Json;
     using System.Threading.Tasks;
 
-    public class NuGetProjectManager : TypedManager<NuGetPackageVersionMetadata>
+    public class NuGetProjectManager : TypedManager<NuGetPackageVersionMetadata, NuGetProjectManager.NuGetArtifactType>
```

Comment:
Is there a case where we would have non-matching package metadata and enum type here? should the enum be defined in the meta data class and then referenced by the manager so it doesn't require two type params?

Reply:
I could merge them as a part of the larger metadata refactoring work, which I just started working on today.

Reply:
I think that makes sense to do before converting everything to typed managers. I cannot think of a case where you'd want ones that don't match very closely.

Reply:
Tracking this work with #333